src/btree/bt_compact.c

```diff
@@ -103,6 +103,37 @@ __compact_rewrite_lock(WT_SESSION_IMPL *session, WT_REF *ref, bool *skipp)
 	return (ret);
 }
 
+/*
+ * __compact_progress --
+ *     Output a compact progress message.
+ */
+static void
+__compact_progress(WT_SESSION_IMPL *session)
+{
+	struct timespec cur_time;
+	WT_BM *bm;
+	uint64_t time_diff;
+
+	if (!WT_VERBOSE_ISSET(session, WT_VERB_COMPACT_PROGRESS))
+		return;
+
+	bm = S2BT(session)->bm;
+	__wt_epoch(session, &cur_time);
+
+	time_diff = WT_TIMEDIFF_SEC(cur_time, session->compact->begin);
+	if (time_diff / 20 > session->compact->prog_msg_count) {
```

Comment:
I'd like a comment here because it isn't obvious what the connection between a time difference and message count is. Something like: "Wait 20 seconds before printing out any progress messages and then print out one message every 20 seconds."
It seems like the `20` should be a macro because that time might want to be adjusted.